tasks/upload.yml

```diff
@@ -95,6 +99,7 @@
     ramdisk: "{{ item.2.id if is_baremetal else omit }}"
   vars:
     is_baremetal: "{{ item.0.elements is defined and 'baremetal' in item.0.elements }}"
+    visibility: "{{ 'public' if item.0.is_public else 'private' if item.0.is_public == False else item.0.visibility }}"
```

Comment:
Does this will blow up if is_public is not defined?

Also drop the `bool` filter.

I might do it like this:

```
visibility: "{{ item.0.visibility | default(item.0.is_public | ternary('public', 'private') if item.0.is_public is defined else os_images_visibility) }}"
```

Reply:
done